File: algorithms/rs/26.删除有序数组中的重复项.rs
/*
 * @lc app=leetcode.cn id=26 lang=rust
 *
 * [26] 删除排序数组中的重复项
 */

// @lc code=start
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let len = nums.len();
        if len < 2 {
            return len as i32;
        }
        let mut i = 0;
        for j in 1..len {
            if nums[i] != nums[j] {
                if (j - i > 1) {
                    nums[i + 1] = nums[j];
                }
                i += 1;
            }
        }
        (i + 1) as i32
    }
}
// @lc code=end

File: algorithms/golang/26.删除有序数组中的重复项.go
/*
 * @lc app=leetcode.cn id=26 lang=golang
 *
 * [26] 删除有序数组中的重复项
 */

// @lc code=start
func removeDuplicates(nums []int) int {
	size := len(nums)
	if size < 2 {
		return size
	}
	n := 1
	for i := 1; i < size; i++ {
		if nums[i] == nums[n-1] {
			continue
		} else {
			nums[n] = nums[i]
			n++
		}
	}
	return n
}

// @lc code=end

